source/adapters/level_zero/queue.cpp

```diff
@@ -1441,8 +1471,19 @@ ur_result_t ur_queue_handle_t_::synchronize() {
               UR_CALL(syncImmCmdList(this, ImmCmdList));
           } else {
             for (auto &ZeQueue : QueueGroup.second.ZeQueues)
-              if (ZeQueue)
-                ZE2UR_CALL(zeHostSynchronize, (ZeQueue));
+              if (ZeQueue) {
```

Comment:
This happens 3 times. Maybe this should be a function in Queue?

Reply:
I checked the templating and it seems to add to much complexity for this case. I left this out to make the change simpler for now.